src/hostapi/wasapi/pa_win_wasapi.c

```diff
@@ -5849,7 +5930,9 @@ PA_THREAD_FUNC ProcThreadEvent(void *param)
         return (UINT32)paUnanticipatedHostError;
 
     // Request fine (1 ms) granularity of the system timer functions for precise operation of waitable timers
-    SystemTimer_SetGranularity(&timer, 1);
+    // Note: while granularity is already set by OpenStream() we still must be sure that user code did not
```

Comment:
This comment does not make sense to me. Once PA/WASAPI has reduced the timer granularity, it would be very difficult for client code to increase the granularity. Client code would have to issue an unmatched call to `timeEndPeriod` (already a bug in client code) with exactly the same granularity that PA/WASAPI used to set the granularity. Under these circumstances "we still must be sure" seems vague and confusing. The comment could say "while granularity has already been set by OpenStream(), we defensively set it again here to avoid issues with buggy client code." Then at least it is clear why you are doing it, but without further explanation I still see no likelihood of it being needed. It would be good for the comment to say why this could actually happen, especially if you have seen this in practice.

P.S. I checked and see that all of the Set/Restore calls are matched, so I resolved my earlier comment.

Reply:
> Once PA/WASAPI has reduced the timer granularity, it would be very difficult for client code to increase the granularity.

My assumption was that usually `Pa_Initialize` is called somewhere early when app starts. Then user code can do some work and accidentally change granularity. In this case PA WASAPI is making sure that when stream is opened the granularity is on required level. In any case PA WASAPI is matching calls correctly, therefore I see no potential issue with a client app. This approach can be treated as defensive.

Reply:
To be clear: I don't think the code actually does anything because of the way windows timer API works: it will always choose the minimum of all requested granularities so it is very hard for the client to interfere with PortAudio's previous requests. So, I would like the code reviewed.

However, if you insist on retaining the code, then the comment needs to clearly explain why the code actually could have some effect. The comment right now is confusing because it contradicts the basic mechanism that the windows timer API uses.

Reply:
@RossBencina you are right. I checked Microsoft docs for [timeBeginPeriod](https://learn.microsoft.com/en-us/windows/win32/api/timeapi/nf-timeapi-timebeginperiod) telling that _"For processes which call this function, Windows uses the lowest value (that is, highest resolution) requested by any process."_

I updated implementation and `SystemTimer_SetGranularity()` is called in once in `OpenStream()`.